redaxo/src/core/functions/function_rex_escape.php

```diff
@@ -12,13 +11,16 @@
  * @param mixed  $value    The value to escape
  * @param string $strategy Supported strategies:
  *                         "html": escapes a string for the HTML context.
+ *                         "html_simplified": escapes a string for the HTML context. Allows some basic tags which are safe regarding XSS.
```

Comment:
Ganz optimal finde ich den Namen nicht. Aber ggf. lassen wir auch mal fünfe gerade sein. ;)

Aber die strategy bezeichnet im Wesentlichen ja, für welchen Kontext escaped wird. Also für HTML-Kontext, oder für HTML-Attribut-Kontext, oder JS-Kontext etc.
Und hier meint es jetzt ja eher, was rauskommt: einfaches HTML. Und nicht, in welchen Kontext es eingesetzt wird (es wird nicht in einfaches HTML eingesetzt, sondern in ganz normales HTML, und erlaubt dabei trotzdem einfaches HTML innerhalb). Somit ist es für mich genau genommen eigentlich eher `html_complex` (nicht ernstgemeinter Vorschlag), also für HTML-Kontext, in komplexerer Variante.

Aber lassen wir mal gut sein. Vermutlich denke ich hier zu genau ;) Und dein Vorschlag ist kürzer als mein `html_allow_simple_tags`.

Reply:
ja ich finds auch gut langsam :-).

einzige frage die mir noch im kopf rummgeistert: brauchen wir überhaupt dieses html_simplified, oder sollte man ggf. einfach nur `rex_string::sanitizeHtml` nehmen.. dort sind die tags aus dem simplified vermutlich auch erlaubt und womöglich auch noch sicherer implementiert

Reply:
> womöglich auch noch sicherer implementiert

Da habe ich Zweifel. Ich finde unsere Implementierung hier sehr übersichtlich, und behaupte, sie ist sicher.
AntiXSS hingegen scheint mir ein One-Man-Projekt zu sein, und enthält recht komplexe Logik (so scheint mir).
Ich habe schon das Gefühl, dass der Typ weiß, was er da tut. Allerdings ändert er ja auch regelmäßig Dinge und erweitert die Logik noch. Daher habe ich schon bisschen Respekt vor dem Ding, ob das immer so wirklich korrekt ist.

Hinzu kommt aber noch ein anderer Aspekt: saniziteHtml geht von einer anderen Basis aus: dass es grundsätzlich HTML ist, und dabei unsichere Dinge herausgefiltert werden sollen. `<meinText>` zum Beispiel ist aber nicht unsicher, und bleibt wahrscheinlich stehen. Und würde dann als HTML-Tag verschluckt werden.
`rex_escape` hingegen geht davon aus, dass es grundsätzlich erst mal kein HTML ist, und deswegen HTML-Sondersymbole escaped werden sollen, damit im HTML-Kontext der Text dann korrekt erscheint.
Oder anders: bei sanitizeHtml müssten wir in den lang-dateien überall `&quot;`, `&amp;` etc. direkt notieren.

Reply:
Noch ein Aspekt: Ich vermute, dass es ein nicht völlig unerheblicher Performance-Unterschied wäre. Gerade weil hier alle Übersetzungsstrings einzeln durchgehen.